server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -596,6 +601,17 @@ public void test_eq_on_bool_uses_termquery() throws Exception {
         assertThat(query).isExactlyInstanceOf(TermQuery.class);
     }
 
+    @Test
+    public void test_eq_on_alias_uses_termquery() throws Exception {
+        // Testing expression: col as alias = 'foo'
+        AliasSymbol alias = new AliasSymbol("aliased", createReference("name", DataTypes.STRING));
+        var literal = Literal.of("foo");
+        var func = new Function(EqOperator.SIGNATURE, List.of(alias, literal), DataTypes.BOOLEAN);
+        Query query = queryTester.toQuery(func);
+        assertThat(query).isNotExactlyInstanceOf(GenericFunctionQuery.class);
+        assertThat(query.toString()).doesNotContainIgnoringCase("aliased");
+    }
```

Comment:
We cannot restict the type of Lucene query generated, as long as it is not a slow GenericFunctionQuery, it should be ok. Also making sure that `aliased` is gone. Both asserts fail if EvaluatingNormalizer is turned off.